@sthibaul See the CI failures, like https://gitlab.spack.io/spack/spack-packages/-/jobs/20696760.
```
checking termcap support using ncurses and ... no
checking termcap support using ncurses and -lncursesw... no
checking termcap support using ncurses and -lncurses... no
checking termcap support using ncurses and -ltinfo... yes
```
and
```
/usr/bin/ld: lstopo_no_graphics-lstopo-ascii.o: in function `output
            _ascii':
  >> 659    /tmp/root/spack-stage/spack-stage-hwloc-2.13.0-5jigahsf3hy6dgzsbgd2
            wjukdw3g57o4/spack-src/utils/lstopo/lstopo-ascii.c:610: undefined r
            eference to `wrefresh'
  >> 660    collect2: error: ld returned 1 exit status
```
With `^ncurses+termlib`, the link line only adds `-lm -ltinfo` instead of `-lm -lncurses -ltinfo`.

--- a/repos/spack_repo/builtin/packages/hwloc/package.py
+++ b/repos/spack_repo/builtin/packages/hwloc/package.py
@@ -131,6 +131,13 @@ class Hwloc(AutotoolsPackage, CudaPackage, ROCmPackage):
     depends_on("ncurses ~termlib", when="@2.0:2.2")
     depends_on("ncurses ~termlib", when="@1.0:1.11.12")
 
+    # Removes unnecessary refresh call that fails linking with ^ncurses+termlib
+    patch(
+        "https://github.com/open-mpi/hwloc/commit/f7f1f76573ce505dae73568c912d2b2efdbf0f71.patch?full_index=1",
+        sha256="b4db98b39733435273e57b8229ee834ce50d2785641d1587d8039598752b1a3d",
+        when="@2.13.0",
+    )
+
     # When mpi=openmpi, this introduces an unresolvable dependency.
     # See https://github.com/spack/spack/issues/15836 for details
     depends_on("mpi", when="+netloc")

